_posts/2020-08-31-final-report-yuxuan.md

```diff
@@ -0,0 +1,169 @@
+
+- Combine the results
+
+```python
+result = np.concatenate(results)
+result = np.asarray(result).T
+
+labels = ['all', 'x-axis', 'y-axis', 'z-axis']
+for col, label in zip(result, labels):
+    plt.plot(col, label=label)
+plt.legend()
+plt.ylabel('Radius of gyration (Å)')
+plt.xlabel('Frame')
+```
+
+![Radius of gyration results](/public/images/final_report_yuxuan/result.png)
+
+A detailed UserGuide on how to parallelizing your analysis scripts can also be found here: [https://yuxuanzhuang.github.io/UserGuide/examples/analysis/custom_parallel_analysis.html](https://yuxuanzhuang.github.io/UserGuide/examples/analysis/custom_parallel_analysis.html). 
+
+### Future of parallel MDAnalysis
+
+Of course, we don't want to limit ourselves to some simple scripts. In [Parallel MDAnalysis](https://github.com/MDAnalysis/pmda) (PMDA), we are trying out what might be possible (PR #[128](https://github.com/MDAnalysis/pmda/pull/128), #[132](https://github.com/MDAnalysis/pmda/pull/132), #[136](https://github.com/MDAnalysis/pmda/pull/136) with this new feature to build a better parallel AnalysisBase API for the users. After PR #[136](https://github.com/MDAnalysis/pmda/pull/136), you will be able to use native features from dask to build complex analysis tasks and run multiple analyses in parallel as well:
+
+```python
+u = mda.Universe(TPR, XTC)
+ow = u.select_atoms("name OW")
+D = pmda.density.DensityAnalysis(ow, delta=1.0)
+
+# Option one (
+D.run(n_blocks=2, n_jobs=2)
+
+#  Option two
+D.prepare(n_blocks=2)
+D.compute(n_jobs=2)   #  or dask.compute(D)
+
+#  furthermore
+dask.compute(D_1, D_2, D_3, D_4...)  #  D_x as an individual analysis job.
+```
+
+## Conclusions
+
+These three months of GSoC project have been fun and fruitful. Thanks to all my mentors, @IAlibay, @fiona-naughton, @orbeckst, and @richardjgowers, (also @kain88-de and other developers) for their help, reviewing PRs, and providing great advice for this project. Now MDAnalaysis has a more streamlined way to build parallel code for the trajectory analysis. What's left to do is reaching a consensus on what the future parallel analysis API should look like, minimizing the memory usage and time for the serialization process, and conducting conclusive benchmarks on different parallel engines 
+
+## Appendix
+
+### Benchmark
+The benchmark below was run with four AMD Opteron 6274 (64 cores in total). The test trajectories (9000 frames, 111815 atoms) were available at DOI [10.6084/m9.figshare.8202149](https://figshare.com/articles/Raw_data/6025748).
+
+Three test cases were conducted under the latest MDAnalysis and PMDA code (PR #136)---RMSD analysis,  a more compute-intensive RDF analysis, and the newly supported RMSD analysis with on-the-fly transformation (fit-rot-trans transformation in this case). All these cases show a strong scaling performance before 16 cores. 
+
+![Benchmark](/public/images/final_report_yuxuan/benchmarking.png)
+
```

Comment:
By the way, it's a bit sad that the RMSD on-the-fly transformation seems to be more than twice as slow as just the RMSD calculation. I don't know why this would be the case. 

You don't have to do or say anything about it, I just wanted to comment on it before I forget.